UM/PluginRegistry.py

```diff
@@ -397,14 +395,23 @@ def loadPlugin(self, plugin_id: str) -> None:
             except InvalidMetaDataError:
                 return
 
-        #If API version is incompatible, don't load it.
-        if self._metadata[plugin_id].get("plugin", {}).get("api", 0) != self.APIVersion:
+        # If API version is incompatible, don't load it.
+        plugin_api_version = self._metadata[plugin_id].get("plugin", {}).get("api", Version("0"))
+
+        if plugin_api_version.getMajor() != self.APIVersion.getMajor():
             Logger.log("w", "Plugin %s uses an incompatible API version, disabling", plugin_id)
             self.disablePlugin(plugin_id)
             return
 
+        if plugin_api_version > self.APIVersion:
```

Comment:
@ChrisTerBeke 
According to the SDK versioning document in confluence:

5.2 → 5.2.0 | 5.1 → 5.1.0 | --> The major number is the same, and the minor number requirement is lower can currently available, so we load it.

I changed the plugin "Console Logger" API to 5.2 and   "self.APIVersion" to 5.1. The plugin is loaded but disabled. I think we need to update the document.

Reply:
No, we should fix the behavior in Cura then. A plugin 'created' for 5.1 should load fine in 5.2 as that version change should only include new functionality and no breaking changes (as per SemVer specification).

Reply:
If Cura has plugin version 5.1 and the plugin needs 5.2 to work, it makes total sense that it doesn't work... 

Cura 5.1, plugin 5.0 (Should work)
Cura 5.1, plugin 5.1 (Should work)
Cura 5.1, plugin 5.2 (Should not work) 
Cura 6.0, plugin 5.2 (Should not work (different major version nr))

As far as I can see, the code does exactly that.

Reply:
Ah wait, @nallath is correct. The other way around should work, but it should not load when Cura is at a lower version than the plugin requires.

Reply:
Ah, I found the issue; It used to disable plugins if they had a wrong version number at one point. After that, they never got re-evaluated. I've fixed that now.

Reply:
I moved the version compatibility check into a separate function and added unit tests for it in 
14c7523